onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -1825,3 +1825,35 @@ MlasNhwcAvgPool(
     );
 
 #endif
+
+struct MlasFlashAttentionThreadedArgs {
+    int batch_size;
+    int num_heads;
+    int q_sequence_length;
+    int kv_sequence_length;
+    int qk_head_size;
+    int v_head_size;
+    int row_size_q;
+    int row_size_kv;
+    float scale;
+    int thread_count;
+    float* buffer;
+    size_t buffer_size_per_thread;
+    const float* query;
+    const float* key;
+    const float* value;
+    float* output;
+};
+
+/**
+ * @brief Per-thread worker function for fp32 Flash Attention
```

Comment:
what is the reason to make the function call per-thread? Is it better to have a MlasFlashAttention with multiple-threading support from the API perspective?

Reply:
Moved multi-threading inside the MLAS kernel. Please have a look again.